### DISCUSSION TITLE
Can MOOSE work with overlapping subdomains?

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#19501] [Q&A General] [answered]
Hi guys, I'm having trouble uploading Abaqus mesh (with overlapping subdomain) to MOOSE

MOOSE's `meshfile` and `FileMeshGenerator` don't seem to detect the subdomain within the larger subdomain for example the figure below (mining-related)

the green and the yellow are the model rock type block and the smaller blocks are the mining activity blocks,
but in this case, MOOSE will only see the rock type blocks and completely ignore the smaller blocks 

I need both of them for my workflow, is there a way to deal with this issue?
![image](https://user-images.githubusercontent.com/79562439/143963539-afd5e6b4-1615-4ad4-bea2-3ed061031425.png)

Kind regards,
Traiwit

**Comment by hugary1995** [comment, marked as answer]
Every element belongs to one and only one subdomain. This has been a fundamental assumption in lots of code in libmesh. You may look into alternatives such as elemental auxvariables and extra element IDs.
